packages/superset-ui-control-utils/src/InfoTooltipWithTrigger.tsx

```diff
@@ -0,0 +1,83 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { kebabCase } from 'lodash';
+// @ts-ignore
+import { Tooltip, OverlayTrigger } from 'react-bootstrap';
+
+const defaultProps = {
+  className: 'text-muted',
+  icon: 'info-circle',
+  placement: 'right',
+};
+const tooltipStyle = { wordWrap: 'break-word' } as any;
+
+interface Props {
+  label: string;
+  tooltip: string;
+  icon: string;
+  onClick: () => void;
+  placement: string;
+  bsStyle: string;
+  className: string;
+}
```

Comment:
Make those with defaults optional.

```ts
interface Props {
  label: string;
  icon: string;
  tooltip?: string;
  onClick?: () => void;
  placement?: string;
  bsStyle?: string;
  className?: string;
}
```

Reply:
+1 @ktmud with the small change to use `type` for `Props` instead of `interface`. (Just convention from guidelines at Airbnb and used throughout the codebase at the moment)

Reply:
`interface` means the properties are immutable which often runs into problems when you want to extend the component and override the prop types.